src/models/state/mod.rs

```diff
@@ -1233,11 +1259,19 @@ impl GlobalState {
                 }
 
                 // apply removals
-                for removal_record in &removals {
-                    membership_proof.update_from_remove(removal_record);
-                    block_msa.remove(removal_record);
+                let mut applied_removals = removals.clone();
```

Comment:
this clone is unnecessary because removals is not used after this.   So it can be:
```
let mut applied_removals = removals;
```

or just use `removals` in the loop.